### PR TITLE
[KOGITO-2150][KOGITO-2117] Adjust deploy pipeline & Setup promote pipeline

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-2150
https://issues.redhat.com/browse/KOGITO-2117

`Jenkinsfile.nightly` will be deprecated and replaced by `Jenkinsfile.deploy` (which is a copy of it with some improvements/parameters)
With this PR, we will need also to update the Jenkins job with the new file.
Finally default deployment registry is done Openshift.

The parent nightly pipeline will first call the `Jenkinsfile.deploy` pipeline and then if everything is ok, will call the `Jenkinsfile.promote` pipeline.
https://github.com/kiegroup/kogito-pipelines/pull/8

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: Jenkinsfile.deploy [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        string(name: 'IMAGE_NAME_SUFFIX', defaultValue: '', description: 'Image name suffix to use to deploy images. In case you need to change the final image name, you can add a suffix to it.')
+        string(name: 'IMAGE_TAG', defaultValue: '', description: 'Image tag to use to deploy images')
+
+        // To define if needed. If not, default config will apply
+        booleanParam(name: 'KOGITO_IMAGES_IN_OPENSHIFT_REGISTRY', defaultValue: false, description: 'Set to true if kogito images for tests are in internal Openshift registry.In this case, KOGITO_IMAGES_REGISTRY and KOGITO_IMAGES_NAMESPACE parameters will be ignored')
+        string(name: 'KOGITO_IMAGES_REGISTRY', defaultValue: '', description: 'Test images registry')
+        string(name: 'KOGITO_IMAGES_NAMESPACE', defaultValue: '', description: 'Test images namespace')
+        string(name: 'KOGITO_IMAGES_NAME_SUFFIX', defaultValue: '', description: 'Test images name suffix')
+        string(name: 'KOGITO_IMAGES_TAG', defaultValue: '', description: 'Test images tag')
+        string(name: 'EXAMPLES_REF', defaultValue: '', description: 'Test image registry')
```
[the remark this answers is not in the document]
Comment (radtriste) [reply Jenkinsfile.deploy]: good catch, will update it

File: Jenkinsfile.deploy [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+                }
+            }
+        }
+        stage('Initialize') {
+            steps {
+                script {
+                    sh "mkdir -p ${WORKING_DIR} && cd ${WORKSPACE} && cp -Rap * ${WORKING_DIR}"
+                    sh "set +x && oc login --token=\$(oc whoami -t) --server=${OPENSHIFT_API} --insecure-skip-tls-verify"
+                    
+                    setupImageEnvVarsFromParams("","DEPLOY")
```
[the remark this answers is not in the document]
Comment (radtriste) [reply Jenkinsfile.deploy:59]: This is setting the `DEPLOY_*` images environment variables based on the input.
I want to keep separated `params` and `env` to have a clear view of what we need.
I will add the needed env vars into the `environment` part

[the remark this answers is not in the document]
Comment (spolti) [reply Jenkinsfile.deploy:71]: ok

File: Jenkinsfile.deploy (hunk 1 above)
[the remark this answers is not in the document]
Comment (spolti) [reply Jenkinsfile.deploy]: looks like you forget to update it.

[the remark this answers is not in the document]
Comment (radtriste) [reply Jenkinsfile.deploy]: updated :)

**Comment by radtriste** [comment]
@sutaakar @Sgitario could you have a final look before I merge this one ?

**Pull request merged**